code/mathematical-algorithms/factorial/FastFactorialUsingStack.java

```diff
@@ -0,0 +1,135 @@
+import java.util.*;
+import java.io.InputStreamReader;
+import java.io.BufferedReader;
+/*
+	@author Kanika Saini (https://github.com/kanikasaini)
+*/
+class mystack implements Cloneable //stack class
```

Comment:
Class name violates naming convention. It should be MyStack (or even better Stack) instead.

Reply:
Also are you sure you need your own implementation of data structure? Why not use java.util.Stack?